synapse/lib/auth.py

```diff
@@ -895,6 +909,8 @@ async def setRules(self, rules, gateiden=None, nexs=True, mesg=None):
     async def addRule(self, rule, indx=None, gateiden=None, nexs=True):
         s_schemas.reqValidRules((rule,))
         rules = self.getRules(gateiden=gateiden)
+        if rule in rules:
```

Comment:
This has a subtle behavior change.

Currently if you had two rules, `A` and `B`; and wanted to swap them, you can do `addRule(A)` and then `delRule(A)`. It's not efficient ( that is what `setRules()` is for), but it would work; since we would append `A` and then remove the earlier `A` rule. Existing code that does that would result in only having rule `B` on it.